### PR TITLE
Module to map tax product

### PR DESCRIPTION
**Pull request by albertofl**
Migraton for bazaar

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Alberto, please don't include pyc files.

Regards.

**Comment by albertofl** [comment]
I have deleted pyc files. Thanks.

**Comment by bwrsandman** [comment]
@albertofl, if you can, squash those two commits

File: magentoerpconnect_product_tax/product.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class account_tax(orm.Model):
+    _inherit = 'account.tax'
+
+    _columns = {
+        'magento_tax_id': fields.integer('Magento Tax ID'),
+    }
+
+
+@magento1700
```
Comment (guewen) [inline magentoerpconnect_product_tax/product.py:17]: It should use the `replacing` syntax (http://odoo-connector.com/api/api_backend.html?highlight=replacing#connector.backend.Backend.parent), but it should not replace directly `ProductImportMapper`, instead, it should replace a Mapper dedicated for the taxes, like it has been done here: https://github.com/OCA/connector-magento/blob/7.0/magentoerpconnect/product.py#L435

File: magentoerpconnect_product_tax/product.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @mapping
+    def tax_id(self, record):        
+        tax_class_id = record.get('tax_class_id', '-1')       
+        sess = self.session
+        tax_ids = sess.search('account.tax',[('magento_tax_id','=',tax_class_id)])
+        if tax_ids:        
+            result = {'taxes_id': [(6, 0, tax_ids)]}
+        else:
+            result = {'taxes_id': []}
```
Comment (guewen) [inline magentoerpconnect_product_tax/product.py:30]: If nothing is found, I think it should return `None`, then, the tax configured on the product will be applied.

File: magentoerpconnect_product_tax/magento_model.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#         versions = super(magento_backend, self)._select_versions(cr, uid, context=context)
+#         versions.append(('1.7-myversion', '1.7 My Version'))
+#         return versions
+
+#     _columns = {
+#         'version': fields.selection(
+#             _select_versions,
+#             string='Version',
+#             required=True),
+#         }
```
Comment (guewen) [inline magentoerpconnect_product_tax/magento_model.py:22]: file to remove

File: magentoerpconnect_product_tax/connector.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier
+#    Copyright 2013 Camptocamp SA
```
Comment (guewen) [inline magentoerpconnect_product_tax/connector.py:5]: wrong author/copyright

File: magentoerpconnect_product_tax/product.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (guewen) [inline magentoerpconnect_product_tax/product.py:1]: missing copyright and license

File: magentoerpconnect_product_tax/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Magento Connector Product Tax',
+    'version': '0.1',
+    'category': 'Connector',
+    'description': """
+    Map Tax in Import
```
Comment (guewen) [inline magentoerpconnect_product_tax/__openerp__.py:27]: Description could be better, with a how-to to configure the taxes ids

File: magentoerpconnect_product_tax/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier
+#    Copyright 2013 Camptocamp SA
```
Comment (guewen) [inline magentoerpconnect_product_tax/__openerp__.py:5]: wrong author and copyright

File: magentoerpconnect_product_tax/__init__.py [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+
+import connector
+#import backend
+#import magento_model
```
Comment (guewen) [inline magentoerpconnect_product_tax/__init__.py:5]: lines to remove

**Comment by guewen** [comment]
Please rebase

File: magentoerpconnect_product_tax/product.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+from openerp.addons.connector.unit.mapper import mapping
+from openerp.addons.magentoerpconnect.product import ProductImportMapper
+from openerp.addons.magentoerpconnect.backend import magento1700
+
+
+class account_tax(orm.Model):
+    _inherit = 'account.tax'
+
+    _columns = {
+        'magento_tax_id': fields.integer('Magento Tax ID'),
```
Comment (OSguard) [inline magentoerpconnect_product_tax/product.py:13]: the magento_tax_id should be related to the magento_backend_id, if you have two magento stores setup the id on magento could be different of each store, so we need to cover this case as well.

**Comment by pedrobaeza** [comment]
@carlosliebana, are you going to work on this?

**Comment by pedrobaeza** [comment]
@liebana, are you going to work on this?

**Comment by pedrobaeza** [comment]
No answer in a while, so closing.

**Pull request closed without merging**